projects/vendor/src/yotpo/baseyotpo/baseyotpo.component.ts

```diff
@@ -0,0 +1,32 @@
+import { OnInit, ElementRef } from '@angular/core';
+import { Observable } from 'rxjs';
+import { Product, ProductService, RoutingService } from '@spartacus/core';
+import { filter, map, switchMap } from 'rxjs/operators';
+
+export abstract class BaseyotpoComponent implements OnInit {
+  product$: Observable<Product>;
+
+  constructor(protected elementRef:ElementRef, 
+		protected routingService: RoutingService,
+		protected productService: ProductService) { }
+
+  ngOnInit() {
+    this.product$ = this.getProduct();
+  }
+
+  getProduct(): Observable<Product> {
+    return this.routingService.getRouterState().pipe(
+      map(state => state.state.params['productCode']),
+      filter(productCode => !!productCode),
+      switchMap((productCode: string) => this.productService.get(productCode))
+    );
+  }
+
+  ngAfterViewInit() {
```

Comment:
This is not called at the moment so you can either remove it or your class should implement the AfterViewInit. That said, I'm not sure what it does.

Reply:
fixed